typed-racket-lib/typed-racket/typecheck/toplevel-trampoline.rkt

```diff
@@ -66,22 +66,23 @@
           (local-expand/capture* #'e 'top-level (kernel-form-identifier-list))))
        (syntax-parse head-expanded
          #:literal-sets (kernel-literals)
-         [(begin (define-values (n) _) ...
-                 (~and (~or _:ignore^ _:ignore-some^)
-                       (~not (~or _:tr:class^
-                                  _:tr:unit^
-                                  _:tr:unit:invoke^
-                                  _:tr:unit:compound^
-                                  _:tr:unit:from-context^))))
-          head-expanded]
```

Comment:
I'm not sure what you mean by "drop", but this code is now handled in the `define-values` handling instead of in the `begin` handling -- @takikawa's new code just pushes the syntax properties around and recurs-via-trampoline on the subforms of the `begin`.